rest/.jvm/src/test/scala/io/udash/rest/EndpointsIntegrationTest.scala

```diff
@@ -23,16 +24,21 @@ class EndpointsIntegrationTest extends UdashSharedTest with BeforeAndAfterAll wi
   val port = 44598
   val contextPrefix = "/rest_api"
   val baseUri = s"http://127.0.0.1:$port$contextPrefix"
-  val server = new Server(port)
-  val context = new ServletContextHandler()
-  context.setSessionHandler(new SessionHandler)
-  context.insertHandler(new GzipHandler)
 
-  private val servlet = io.udash.rest.RestServlet[TestServerRESTInterface](new TestServerRESTInterfaceImpl)
-  val holder = new ServletHolder(servlet)
+  val holder = new ServletHolder(io.udash.rest.RestServlet[TestServerRESTInterface](new TestServerRESTInterfaceImpl))
   holder.setAsyncSupported(true)
-  context.addServlet(holder, s"$contextPrefix/*")
+
+  val contextHandler = new ServletContextHandler()
+  contextHandler.setSessionHandler(new SessionHandler)
+  contextHandler.addServlet(holder, s"$contextPrefix/*")
+
+  val context = new GzipHandler()
+  context.setHandler(contextHandler)
+
+  val server = new Server(port)
   server.setHandler(context)
+  // Unsafe URI compliance is required for testing purposes
+  server.getConnectors.head.getConnectionFactory(classOf[HttpConnectionFactory]).getHttpConfiguration.setUriCompliance(UriCompliance.UNSAFE)
```

Comment:
URICompliance in jetty 10 looked more or less like this: 
`new UriCompliance("LEGACY_DEFAULT", JEnumSet(Violation.AMBIGUOUS_PATH_SEPARATOR, Violation.AMBIGUOUS_PATH_ENCODING))`
Are you sure we want to allow more violations?

Reply:
It was just for tests but I'll use the legacy one, thanks.